.github/workflows/main.yml

```diff
@@ -1,42 +1,182 @@
+                --work-dir /tmp/mpbb \
+                list-subports \
+                --archive-site= \
+                --archive-site-private= \
+                "$port" \
+              | tr '\n' ' ')
+            for subport in $new_subports; do
+              echo "$subport"
+              subportlist="$subportlist $subport"
+            done
+            echo "::endgroup::"
+          done
+
+          echo "::set-output name=subportlist::${subportlist}"
+        env:
+          portlist: ${{ steps.portlist.outputs.portlist }}
+      - name: Run port lint for all changed subports
+        run: |
+          set -eu
+
+          fail=0
+          for subport in $subportlist; do
+            echo "::group::${subport}"
+            path=$(port file "$subport")
+            messagetype="warning"
+            if ! messages=$(port -q lint --nitpick "$subport" 2>&1); then
+              messagetype="error"
+              fail=1
+            fi
+            if [ -n "$messages" ]; then
+              echo "$messages"
+              # See https://github.com/actions/toolkit/issues/193#issuecomment-605394935
+              encoded_messages="port lint --nitpick ${subport}:%0A"
+              encoded_messages+="$(echo "${messages}" | sed -E 's/$/%0A/g' | tr -d '\n')"
+              echo "::${messagetype} file=${path#${PWD}/ports/},line=1,col=1::${encoded_messages}"
+            fi
+            echo "::endgroup::"
+          done
+
+          exit "$fail"
+        env:
+          subportlist: ${{ steps.subportlist.outputs.subportlist }}
+      - name: Build changed subports
+        run: |
+          set -eu
+
+          fail=0
+          for subport in $subportlist; do
+            workdir="/tmp/mpbb/$subport"
+            mkdir -p "$workdir/logs"
+            touch "$workdir/logs/dependencies-progress.txt"
+
+            echo "::group::Cleaning up between ports"
+            sudo mpbb --work-dir "$workdir" cleanup
+            echo "::endgroup::"
+
+            echo "::group::Installing dependencies for ${subport}"
+            sudo mpbb \
+              --work-dir "$workdir" \
+              install-dependencies \
+              "$subport" >"$workdir/logs/install-dependencies.log" 2>&1 &
+            deps_pid=$!
+
+            tail -f "$workdir/logs/dependencies-progress.txt" 2>/dev/null &
+            tail_pid=$!
+
+            set +e
+            wait "$deps_pid"
+            deps_exit=$?
+            set -e
+
+            kill "$tail_pid" || true
+
+            if [ "$deps_exit" -ne 0 ]; then
+              echo "::endgroup::"
+              echo "::error::Failed to install dependencies for ${subport}"
+              continue
+            fi
+            echo "::endgroup::"
+
+            echo "::group::Installing ${subport}"
+
+            set +e
+            sudo mpbb \
+              --work-dir "$workdir" \
+              install-port \
+              --source \
+              "$subport"
+            install_exit=$?
```

Comment:
Before installing the next port, we need to deactivate all ports to detect missing dependencies in the next round.

Reply:
See "Cleaning up between ports", that should do it. I don't see a code path that skips this.